server/konnectors/forex.coffee

```diff
@@ -0,0 +1,135 @@
+americano = require 'americano-cozy'
+request = require 'request'
+moment = require 'moment'
+crypto = require 'crypto'
+async = require 'async'
+xml2js = require 'xml2js'
+
+localization = require '../lib/localization_manager'
+
+# Helpers
+
+log = require('printit')
+    prefix: 'forex'
+    date: true
+
+# Supported ISO 4217 currency codes
+
+CURRENCIES = [
+    'USD', 'JPY', 'GBP', 'AUD', 'CAD', 'CHF', 'BGN', 'BRL', 'CNY', 'CZK', 'DKK',
+    'HKD', 'HRK', 'HUF', 'IDR', 'ILS', 'INR', 'KRW', 'MXN', 'MYR', 'NOK', 'NZD',
+    'PHP', 'PLN', 'RON', 'RUB', 'SEK', 'SGD', 'THB', 'TRY', 'ZAR'
+]
+
+# URLs
+
+dailyUrl = 'https://www.ecb.europa.eu/stats/eurofxref/eurofxref-daily.xml'
+recentUrl = 'http://www.ecb.europa.eu/stats/eurofxref/eurofxref-hist-90d.xml'
+histUrl = 'https://www.ecb.europa.eu/stats/eurofxref/eurofxref-hist.xml'
+
+# Other urls
+# http://www.global-view.com/forex-trading-tools/forex-history/index.html
+# http://www.global-view.com/forex-trading-tools/forex-history/exchange_csv_report.html?CLOSE_1=ON&start_date=1/1/1900&stop_date=03/06/2015&Submit=Get%20Daily%20Stats
+# http://www.global-view.com/forex-trading-tools/forex-history/exchange_csv_report.html?CLOSE_1=ON&CLOSE_2=ON&CLOSE_4=ON&start_date=1/1/1900&stop_date=03/06/2015&Submit=Get%20Daily%20Stats
+# http://www.federalreserve.gov/datadownload/Output.aspx?rel=H10&series=a6a0113179fdeb9c6fbcdc18575ec09c&lastObs=&from=01/01/2000&to=03/07/2015&filetype=csv&label=omit&layout=seriescolumn
+
+# Models
+
+CurrencyRate = americano.getModel 'CurrencyRate',
+    date: Date
+    rates: (x) -> x
```

Comment:
Note: Here, `(x) -> x` is not an array, but a map of currency code to rate (e.g. `USD -> 1.12`). I thought this was better than an array of pairs (e.g. `0 -> { currency: "USD", rate: 1.12 }`).